mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -190,6 +190,15 @@
     MetaData getMetaData(boolean resetRecent, MailboxSession mailboxSession, MessageManager.MetaData.FetchGroup fetchGroup) throws MailboxException;
 
     /**
+     * Offers direct access to UIDs. This allows per implementation optimisation for retrieving the whole UIDs of a mailbox.
+     *
+     * @param session MailboxSession for the user performing the request.
+     * @return Iterator of MessageUids in this mailbox. They are not necessarily sorted.
+     * @throws MailboxException
+     */
+    Iterator<MessageUid> getUids(MailboxSession session) throws MailboxException;
```

Comment:
I had the same thinking, but I don't like the specific search case, as it is non-explicit from an API point of view.

Thus I preferred adding a specific method for it. It seemed to me it was the best design decision.